### ISSUE TITLE
[12.0/13.0] Error receiving materials from a Kit product BoM that have a different UoM than the product

### ISSUE THREAD

**Issue by reinaldoramosarxi** [odoo/odoo#45871]
Hello,

We've noticed that when receiving materials for a Kit manufacturing product (after starting an MO), if the product has one UoM and the materials have another, when validating the stock entrance it will raise the error: 

"The unit of measure defined on the order line doesn't belong to the same category than the unit of measure defined on the product. Please correct the unit of measure defined on the order line or on the product, they should belong to the same category.".

Thank you.

**Comment by Yenthe666** [comment]
@reinaldoramosarxi well, the message tells you what's wrong right? It just a limitation.
@pedrobaeza you agree?

**Comment by pedrobaeza** [comment]
@Yenthe666 not really sure, the devil is on the details. @reinaldoramosarxi please fill all the details of how is the pack configured and so on.

**Comment by reinaldoramosarxi** [comment]
Hello,

I'm sorry, I think I didn't explain it correctly, let me rephrase the issue.

Consider:
- A saleable product that has a kit type BoM that is set to "Units" UoM.
- The BoM has several components that have a purchase route, and at least one of them has a "Meters" UoM.

The issue:
- When selling this Kit, the sale will generate an incoming shipment (or several, depending on how many suppliers are used for the components).
- When validating the incoming shipment that includes one of the components that doesn't have the same UoM as the product, the error will come up. 
- Even changing the product UoM of that component (which shouldn't be done anyway) will raise the error. 

Sorry for explaining it badly before.

Thank you.
